piff/config.py

```diff
@@ -191,3 +191,122 @@ def plotify(config, logger=None):
     for stats in output.stats_list:
         stats.compute(psf,stars,logger=logger)
         stats.write(logger=logger)
+
+def meanify(config, logger=None):
+    """Take Piff output(s), build an average of the FoV, and wirte output average.
+
+    :param config:      The configuration file that defines how to build the model
+    :param logger:      A logger object for logging progress. [default: None]
+    """
+    from .input import Input
+    from .psf import PSF
+    from .star import Star
+    from .output import Output
+    import glob
+    import numpy as np
+    import fitsio
+
+    if logger is None:
+        verbose = config.get('verbose', 1)
+        logger = setup_logger(verbose=verbose)
+
+    for key in ['input', 'output']:
+        if key not in config:
+            raise ValueError("%s field is required in config dict"%key)
+    for key in ['file_name']:
+        if key not in config['input']:
+            raise ValueError("%s field is required in config dict output"%key)
+
+    for key in ['file_name']:
+        if key not in config['output']:
+            raise ValueError("%s field is required in config dict output"%key)
+
+    if 'dir' in config['input']:
+        dir = config['input']['dir']
+    else:
+        dir = None
+
+    if 'binning' in config['output']:
+        if type(config['output']['binning']) is not int:
+            ValueError("type of binning should be integer (%s now)"%type(config['output']['binning']))
+        else:
+            binning = config['output']['binning']
+    else:
+        binning = 15
+
+    if isinstance(config['input']['file_name'], list):
+        psf_list = config['input']['file_name']
+        if len(psf_list) == 0:
+            raise ValueError("file_name may not be an empty list")
+    elif isinstance(config['input']['file_name'], str):
+        file_name = config['input']['file_name']
+        if dir is not None:
+            file_name = os.path.join(dir, file_name)
+        psf_list = sorted(glob.glob(file_name))
+        if len(psf_list) == 0:
+            raise ValueError("No files found corresponding to "+config['file_name'])
+    elif not isinstance(config['file_name'], dict):
+        raise ValueError("file_name should be either a dict or a string")
```

Comment:
I don't particularly like the file name structure here.  

The way you have it set up, the user will first run a regular config file first with `input` being the data and `output` being the .piff files.

Then they need to rearrange things so that `input` is the old `output` and `output` is a new file for the averages.

Then the do a third run that looks mostly like the first run, but now has an additional `model.average_fits` field.

I think I'd rather let the user use the same config file or at least as close as possible, in all three cases.  And even better if we can combine the first two steps into one.

I'm thinking maybe have a new top-level field, not in output or model or anything, called `hyper`, which would store hyper parameters that use information from multiple exposures.  Then `meanify` would look for an existing set of output files, make them if not present (using the normal piffify function), and then write out to `hyper.file_name` as an output.
Then the GP stuff would look for a `hyper` section and if there is one, get the mean parameters from there to subtract off.

The one thing I think we'd need to add to the regular processing is the ability to specify multiple output files to run (and correspondingly multiple input exposures).  That's not too hard using the GalSim config processing stuff that we are already using to loop over multiple images.  But you don't need to try to do that.  I'll work on that on a separate branch.

For now, I think, just rearrange what you have so that the input to meanify is the `output` field, since that's where it will be in the normal config file.  And its output file name is in a `hyper` field.  Then have that be where GPInterp2pcf will look for the name of the average file.

Then I can tackle making this a bit more seamless for the user to build and run on a set of exposures.

Reply:
Ok, for the moment like Mike suggests, I replaced the `input` field by `output`, and the `output` field by `hyper`